interpro7dw/interpro/utils.py

```diff
@@ -25,16 +25,25 @@ def overlaps_pdb_chain(locations: list[dict], segments: list[dict]) -> bool:
     :param segments:
     :return:
     """
-    for loc in locations:
-        # We do not consider fragmented matches
-        loc_start = loc["fragments"][0]["start"]
-        loc_end = max([f["end"] for f in loc["fragments"]])
 
-        for segment in segments:
-            seg_start = segment["protein_start"]
-            seg_end = segment["protein_end"]
+    def overlaps_pdb_chain(locations: list[dict], segments: list[dict]) -> int:
+        """Evaluate in protein matches and chain segments overlap.
 
-            if loc_start <= seg_end and seg_start <= loc_end:
-                return True
+        :param locations:
+        :param segments:
+        :return:
+        """
+        max_overlap = 0
 
-    return False
+        for loc in locations:
+            # We do not consider fragmented matches
+            loc_start = loc["fragments"][0]["start"]
+            loc_end = max([f["end"] for f in loc["fragments"]])
+
+            for segment in segments:
+                seg_start = segment["protein_start"]
+                seg_end = segment["protein_end"]
+                overlap = (min(loc_end, seg_end) - max(loc_start, seg_start)) + 1
```

Comment:
```suggestion
                overlap = min(loc_end, seg_end) - max(loc_start, seg_start) + 1
```

We don't need these parentheses.